start/common/src/main/java/org/ethereum/beacon/start/common/NodeLauncher.java

```diff
@@ -0,0 +1,321 @@
+import org.ethereum.beacon.wire.WireApiSub;
+import org.ethereum.beacon.wire.WireApiSync;
+import org.ethereum.beacon.wire.WireApiSyncServer;
+import org.ethereum.beacon.wire.message.SSZMessageSerializer;
+import org.ethereum.beacon.wire.net.ConnectionManager;
+import org.ethereum.beacon.wire.sync.BeaconBlockTree;
+import org.ethereum.beacon.wire.sync.SyncManagerImpl;
+import org.ethereum.beacon.wire.sync.SyncQueue;
+import org.ethereum.beacon.wire.sync.SyncQueueImpl;
+import reactor.core.publisher.DirectProcessor;
+import reactor.core.publisher.Flux;
+import reactor.core.publisher.Mono;
+import tech.pegasys.artemis.util.uint.UInt64;
+
+public class NodeLauncher {
+  private final BeaconChainSpec spec;
+  private final DepositContract depositContract;
+  private final List<BLS381Credentials> validatorCred;
+  private final BeaconChainStorageFactory storageFactory;
+  private final Schedulers schedulers;
+
+  private InitialStateTransition initialTransition;
+  private StateCachingTransition stateCachingTransition;
+  private PerSlotTransition perSlotTransition;
+  private PerBlockTransition perBlockTransition;
+  private PerEpochTransition perEpochTransition;
+  private ExtendedSlotTransition extendedSlotTransition;
+  private EmptySlotTransition emptySlotTransition;
+  private BeaconBlockVerifier blockVerifier;
+  private BeaconStateVerifier stateVerifier;
+
+  private InMemoryDatabase db;
+  private BeaconChainStorage beaconChainStorage;
+  private MutableBeaconChain beaconChain;
+  private SlotTicker slotTicker;
+  private ObservableStateProcessor observableStateProcessor;
+  private BeaconChainProposer beaconChainProposer;
+  private BeaconChainAttesterImpl beaconChainAttester;
+  private MultiValidatorService beaconChainValidator;
+
+  private byte networkId = 1;
+  private UInt64 chainId = UInt64.valueOf(1);
+  private boolean startSyncManager = false;
+
+  private WireApiSub wireApiSub;
+  private WireApiSync wireApiSyncRemote;
+  private final ConnectionManager<?> connectionManager;
+  private SimplePeerManagerImpl peerManager;
+  private BeaconBlockTree blockTree;
+  private SyncQueue syncQueue;
+  private SyncManagerImpl syncManager;
+  private WireApiSyncServer syncServer;
+
+  public NodeLauncher(
+      BeaconChainSpec spec,
+      DepositContract depositContract,
+      List<BLS381Credentials> validatorCred,
+      ConnectionManager<?> connectionManager,
+      BeaconChainStorageFactory storageFactory,
+      Schedulers schedulers,
+      boolean startSyncManager) {
+
+    this.spec = spec;
+    this.depositContract = depositContract;
+    this.validatorCred = validatorCred;
+    this.connectionManager = connectionManager;
+    this.storageFactory = storageFactory;
+    this.schedulers = schedulers;
+    this.startSyncManager = startSyncManager;
+
+    if (depositContract != null) {
+      Mono.from(depositContract.getChainStartMono()).subscribe(this::chainStarted);
+    }
+  }
+
+  void chainStarted(ChainStart chainStartEvent) {
```

Comment:
could we split this method to few smaller readable tasks?

Reply:
Yep, good question. I wanted to return to this when create it